addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerJsonParser.java

```diff
@@ -207,34 +195,20 @@ private static ImageIcon createSvgIcon(URL url) {
         if (url == null) {
             return null;
         }
-        String xmlParser = XMLResourceDescriptor.getXMLParserClassName();
-        SAXSVGDocumentFactory df = new SAXSVGDocumentFactory(xmlParser);
-        SVGDocument doc = null;
-        GraphicsNode svgIcon = null;
-        try {
-            doc = df.createSVGDocument(url.toString());
-        } catch (RuntimeException | IOException re) {
-            // v1 SVGs are unsupported
-            return null;
-        }
-        doc.getRootElement().setAttribute("width", String.valueOf(SIZE));
-        doc.getRootElement().setAttribute("height", String.valueOf(SIZE));
-        UserAgent userAgent = new UserAgentAdapter();
-        DocumentLoader loader = new DocumentLoader(userAgent);
-        GVTBuilder builder = new GVTBuilder();
+        SVGLoader loader = new SVGLoader();
+        SVGDocument svgDocument = null;
+        svgDocument = loader.load(url);
```

Comment:
Initialise directly?

Reply:
Doesn't this mean load can return null?

```
[SVGDocument](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github.weisj.jsvg.parser(SVGLoader.class%E2%98%83SVGLoader~load~Ljava.net.URL;%E2%98%82com.github.weisj.jsvg.SVGDocument) [com](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom).[github](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github).[weisj](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github.weisj).[jsvg](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github.weisj.jsvg).[parser](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github.weisj.jsvg.parser).[SVGLoader](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github.weisj.jsvg.parser(SVGLoader.class%E2%98%83SVGLoader).load(@[NotNull](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github.weisj.jsvg.parser(SVGLoader.class%E2%98%83SVGLoader~load~Ljava.net.URL;@xmlBase!0!-1!0!-1!Ljava.net.URL;!-1!true%7Dorg.jetbrains.annotations.NotNull%E2%98%82org.jetbrains.annotations.NotNull) [URL](eclipse-javadoc:%E2%98%82=wappalyzer/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/com.github.weisj%5C/jsvg%5C/1.2.0%5C/7a1dfd541dc48ca58e706c356d1604971f58db25%5C/jsvg-1.2.0.jar=/gradle_used_by_scope=/main,test=/%3Ccom.github.weisj.jsvg.parser(SVGLoader.class%E2%98%83SVGLoader~load~Ljava.net.URL;%E2%98%82java.net.URL) xmlBase)


@Nullable
```

Reply:
Sorry I forgot to check the return type, just the init change then :)